git/url.go

```diff
@@ -35,21 +35,20 @@ func ParseURL(rawURL string) (u *url.URL, err error) {
 		rawURL = "ssh://" + strings.Replace(rawURL, ":", "/", 1)
 	}
 
-	u, err = url.Parse(rawURL)
+	u, err := url.Parse(rawURL)
 	if err != nil {
-		return
+		return nil, err
 	}
 
-	if u.Scheme == "git+ssh" {
-		u.Scheme = "ssh"
-	}
-
-	if u.Scheme == "git+https" {
```

Comment:
These two blocks are mutually exclusive, so they could either be an `if-else` or a `switch`. I chose `switch` because I wanted to make a clear distinction between this conditional and the early return that follows.